### PR TITLE
FDG-3219: fetching user from system env

### PR DESCRIPTION
**Pull request by ankeshflipkart**

### CODE REVIEW

File: beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  public static String getLoggedInUserFromShell() throws IOException, InterruptedException {
+    Optional<String> sudoUser = Optional.fromNullable(System.getenv().get(SUDO_USER_PROP_KEY));
+    Optional<String> user = Optional.fromNullable(System.getenv().get(USER));
+    if (sudoUser.isPresent()) {
+      return sudoUser.get();
+    } else if (user.isPresent()) {
+      return user.get();
+    }
+    return null;
+  }
```
Comment (guptashailesh92) [inline beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java]: return null doesnt make sense. Shouldnt we throw exception.

File: beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+public class ShellUserFetcher
+{
+
+  public static final String SUDO_USER_PROP_KEY = "SUDO_USER";
+  public static final String USER = "USER";
+
+  public static String getLoggedInUserFromShell() throws IOException, InterruptedException {
+    Optional<String> sudoUser = Optional.fromNullable(System.getenv().get(SUDO_USER_PROP_KEY));
+    Optional<String> user = Optional.fromNullable(System.getenv().get(USER));
+    if (sudoUser.isPresent()) {
```
Comment (guptashailesh92) [inline beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java]: move user in if block

File: beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package org.apache.hive.beeline;
+
+import com.google.common.base.Optional;
+
+import java.io.IOException;
+
+public class ShellUserFetcher
+{
```
Comment (guptashailesh92) [inline beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java]: class name should not be doing anything specific to user. should be renamed.

Comment (guptashailesh92) [reply beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java]: soemthing like SystemVariables

Comment (ankeshflipkart) [reply beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java]: Changed

File: beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java (hunk 2 above)
Comment (ankeshflipkart) [reply beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java]: done

File: beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java (hunk 1 above)
Comment (ankeshflipkart) [reply beeline/src/java/org/apache/hive/beeline/ShellUserFetcher.java]: done

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java [hunk 4]
```diff
@@ -835,7 +835,15 @@ private boolean connectUsingArgs(BeelineParser beelineParser, CommandLine cl) {
 
     driver = cl.getOptionValue("d");
     auth = cl.getOptionValue("a");
-    user = cl.getOptionValue("n");
+    /*
+    Fetching user from system properties to check who is loggedin
+     */
+//    user = cl.getOptionValue("n");
+    try {
+      user = ExtractShellUtils.getLoggedInUserFromShell();
```
Comment (mittalnanu) [inline beeline/src/java/org/apache/hive/beeline/BeeLine.java]: why catch throwable if it is a checked exception from the method? Catch exception and put it as runtime

File: beeline/src/java/org/apache/hive/beeline/ExtractShellUtils.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  public static String getLoggedInUserFromShell() throws IOException, InterruptedException {
+    Optional<String> sudoUser = Optional.fromNullable(System.getenv().get(SUDO_USER_PROP_KEY));
+    if (sudoUser.isPresent()) {
+      return sudoUser.get();
+    }
+    Optional<String> user = Optional.fromNullable(System.getenv().get(USER));
+    if (user.isPresent()) {
+      return user.get();
+    }
+    throw new RuntimeException("Unable to identify User");
```
Comment (mittalnanu) [inline beeline/src/java/org/apache/hive/beeline/ExtractShellUtils.java:22]: if you are anyways sending checked exception, don't have runtime exception with it. Reuse the same checked exception

File: beeline/src/test/org/apache/hive/beeline/TestBeelineArgParsing.java [hunk 6]
```diff
@@ -111,7 +111,7 @@ public void testSimpleArgs() throws Exception {
     String args[] = new String[] {"-u", "url", "-n", "name",
       "-p", "password", "-d", "driver", "-a", "authType"};
     org.junit.Assert.assertEquals(0, bl.initArgs(args));
-    Assert.assertTrue(bl.connectArgs.equals("url name password driver"));
+    Assert.assertTrue(bl.connectArgs.equals("url " + System.getenv().get("USER") +" password driver"));
```
Comment (mittalnanu) [inline beeline/src/test/org/apache/hive/beeline/TestBeelineArgParsing.java:114]: where is the test for sudo user?

File: bin/addJars.hql [hunk 7]
```diff
@@ -0,0 +1 @@
+ADD JAR /usr/share/fk-bigfoot-hivejsonserde/json-serde-1.3-SNAPSHOT-jar-with-dependencies.jar; ADD JAR /usr/share/fk-bigfoot-dimlookup/dimlookup-hive-udf-1.0-SNAPSHOT-jar-with-dependencies.jar;
```
Comment (mittalnanu) [inline bin/addJars.hql:1]: where did these extra jar come from ? hive server will already have them right?

File: bin/fdp-beeline.sh [hunk 8]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+beeline -u "jdbc:hive2://prod-fdphadoop-bheema-zk-0001:2181,prod-fdphadoop-bheema-zk-0002:2181,prod-fdphadoop-bheema-zk-0003:2181,prod-fdphadoop-bheema-zk-0004:2181,prod-fdphadoop-bheema-zk-0005:2181/bigfoot_external_neo;transportMode=http;httpPath=cliservice;serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=hiveserver_processing" -i /usr/local/fdp-infra-hive/bin/addJars.hql
```
Comment (mittalnanu) [inline bin/fdp-beeline.sh]: bigfoot_external_neo has all the data?

**Review by mittalnanu — CHANGES_REQUESTED** [review]

File: beeline/src/test/org/apache/hive/beeline/TestBeelineArgParsing.java (hunk 6 above)
Comment (ankeshflipkart) [reply beeline/src/test/org/apache/hive/beeline/TestBeelineArgParsing.java:114]: Can't set sudo user in System.getEnv its unmodifiable properties.

File: bin/addJars.hql (hunk 7 above)
Comment (ankeshflipkart) [reply bin/addJars.hql:1]: Yeah but the jars should be added when beeline execute, currently hive cli by defaults add these configs.

File: bin/fdp-beeline.sh (hunk 8 above)
Comment (ankeshflipkart) [reply bin/fdp-beeline.sh]: we can mention db name directly or we can remove that as well. its directly connects to that db.

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java (hunk 4 above)
Comment (ankeshflipkart) [reply beeline/src/java/org/apache/hive/beeline/BeeLine.java]: done

File: beeline/src/java/org/apache/hive/beeline/ExtractShellUtils.java (hunk 5 above)
Comment (ankeshflipkart) [reply beeline/src/java/org/apache/hive/beeline/ExtractShellUtils.java:22]: will throw only runtime here.

File: bin/fdp-beeline.sh (hunk 8 above)
Comment (ankeshflipkart) [reply bin/fdp-beeline.sh]: removed db as well.

**Review by mittalnanu — APPROVED** [review]
LGTM

**Review by mittalnanu — APPROVED** [review]
LGTM

**Pull request merged**